class.media-summary.php

```diff
@@ -58,10 +58,15 @@ static function get( $post_id, $blog_id = 0, $args = array() ) {
 			foreach ( $extract['shortcode'] as $type => $data ) {
 				switch ( $type ) {
 					case 'wpvideo':
-						if ( 0 == $return['count']['video'] ) {
+						if ( 0 == $return['count']['video'] ){
+							if ( ! class_exists( 'VideoPress_Video' ) ) {
+								require_once( JETPACK__PLUGIN_DIR . '/modules/videopress/class.videopress-video.php' );
+							}
+							$wpvideo = new VideoPress_Video( $extract['shortcode']['wpvideo']['id'][0] );
+
 							$return['type'] = 'video';
-							$return['video'] = esc_url_raw( 'http://s0.videopress.com/player.swf?guid=' . $extract['shortcode']['wpvideo']['id'][0] . '&isDynamicSeeking=true' );
-							$return['image'] = self::get_video_poster( 'videopress', $extract['shortcode']['wpvideo']['id'][0] );
+							$return['video'] = esc_url_raw( $wpvideo->videos->mp4->url );
+							$return['image'] = $wpvideo->poster_frame_uri;
 							$return['secure']['video'] = preg_replace( '@http://[^\.]+.videopress.com/@', 'https://v0.wordpress.com/', $return['video'] );
```

Comment:
Not sure what you mean here. We are using the new URL and swapping out the `http` part with `https` one.